ts/features/itwallet/machine/credential/context.ts

```diff
@@ -1,39 +1,80 @@
 import { CryptoContext } from "@pagopa/io-react-native-jwt";
 import { AuthorizationDetail } from "@pagopa/io-react-native-wallet";
-
 import {
   IssuerConfiguration,
-  StoredCredential,
   RequestObject,
+  StoredCredential,
   WalletInstanceAttestations
 } from "../../common/utils/itwTypesUtils";
 import { CredentialIssuanceFailure } from "./failure";
 
+/**
+ * The mode for the credential issuance process.
+ * It can be:
+ * - "issuance": for a new credential issuance
+ * - "reissuance": for reissuing an existing credential (for example, when the credential is expired
+ *  or about to expire)
+ * - "upgrade": for upgrading an existing credential to a the new format
+ */
+export type CredentialIssuanceMode = "issuance" | "reissuance" | "upgrade";
```

Comment:
Do you think we need to explicitly track this `issuance` vs `reissuance` mode? What are the advantages it brings?

Reply:
"reissuance" is needed in the `EvaluateFlow` state to know if we need to navigate to the next scree. 
This is a short breakdown of the different modes and what they do:
- `issuance` → starts the flow but skips the initial navigation, needed in the credentials onboarding catalog to display an inline loading indicator
- `reissuance` → does not skip navigation and shows the trust screen with a loading indicator
- `upgrade` → like the reissuance, but it also allow to pass the "reissuing" `operationType` header to the` /credential` endpoint

Does it makes sense to you?

Reply:
Ok thanks, it makes sense.